src/components/atoms/toggle-expand/ToggleExpand.template.ts

```diff
@@ -12,7 +12,7 @@ export type ToggleExpandProps = {
 };
 
 export const toggleExpandTemplate = templateComponentFactory<ToggleExpandProps>({
-  component: ToggleExpand,
+  component: ToggleExpand as any,
```

Comment:
@ThaNarie I think this needs a change in Muban to get this working without `as any`

Reply:
What's the exact error you're getting without the `any`?

Reply:
```TS2322: Type 'ComponentFactory<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }, "toggle-expand">' is not assignable to type 'string | ComponentFactory<any, any> | undefined'.   Type 'ComponentFactory<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }, "toggle-expand">' is not assignable to type 'ComponentFactory<any, any>'.     Type 'ComponentFactory<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }, "toggle-expand">' is not assignable to type 'ComponentReturnValue<TypedProps<any>>'.       Call signature return types '{ readonly name: string; setProps: (props: TypedProps<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }>) => void; ... 4 more ...; __instance: Internal...' and '{ readonly name: string; setProps: (props: TypedProps<any>) => void; readonly props: TypedProps<any>; readonly element: HTMLElement; setup: () => void; dispose: () => void; __instance: InternalComponentInstance; }' are incompatible.         The types of 'setProps' are incompatible between these types.           Type '(props: TypedProps<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }>) => void' is not assignable to type '(props: TypedProps<any>) => void'.             Types of parameters 'props' and 'props' are incompatible.               Property 'isExpanded' is missing in type 'TypedProps<any>' but required in type 'TypedProps<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }>'.  ToggleExpand.ts(21, 5): 'isExpanded' is declared here. templateComponentFactory.d.ts(4, 5): The expected type comes from property 'component' which is declared here on type 'TemplateComponentFactoryOptions<any>'```

Reply:
ERROR in \muban-skeleton\src\components\atoms\toggle-expand\ToggleExpand.template.ts
./src/components/atoms/toggle-expand/ToggleExpand.template.ts
[tsl] ERROR in \Muban\muban-skeleton\src\components\atoms\toggle-expand\ToggleExpand.template.ts(15,3)
      TS2322: Type 'ComponentFactory<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }, "toggle-expand">' is not assignable to type 'string | ComponentFactory<any, any> | undefined'.
  Type 'ComponentFactory<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }, "toggle-expand">' is not assignable to type 'ComponentFactory<any, any>'.
    Type 'ComponentFactory<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }, "toggle-expand">' is not assignable to type 'ComponentReturnValue<TypedProps<any>>'.
      Call signature return types '{ readonly name: string; setProps: (props: TypedProps<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }>) => void; ... 4 more ...; __instance: Internal...' and '{ readonly name: st

ring; setProps: (props: TypedProps<any>) => void; readonly props: TypedProps<any>; readonly element: HTMLElement; setup: () => void; dispose: () => void; __instance: InternalComponentInstance; }' are incompatible.
        The types of 'setProps' are incompatible between these types.
          Type '(props: TypedProps<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }>) => void' is not assignable to type '(props: TypedProps<any>) => void'.
            Types of parameters 'props' and 'props' are incompatible.
              Property 'isExpanded' is missing in type 'TypedProps<any>' but required in type 'TypedProps<{ isExpanded: { type: BooleanConstructor; } & { defaultValue: <U_13 extends boolean | undefined>(value: U_13 extends Primitive ? U_13 : () => U_13) => { ...; } & ... 1 more ... & { ...; }; } & { ...; } & { ...; }; }>'.
 @ ./src/components/block-renderer/BlockRenderer.template.ts 8:0-84 15:39-59
 @ ./src/components/layouts/DefaultLayout.template.ts 6:0-82 10:208-229
 @ ./src/App.template.ts 9:0-84 17:40-61
 @ ./src/server-bundle.ts 18:18-55

Reply:
Solved in newer @muban/muban versions.